src/dataflow/src/source/s3.rs

```diff
@@ -934,24 +944,20 @@ impl SourceReader for S3SourceReader {
                     value: record,
                 }))
             }
-            Some(Some(Err(e))) => {
-                tracing::warn!(
-                    "when reading source '{}' ({}): {}",
-                    self.source_name,
-                    self.id,
-                    e
-                );
-                match e {
-                    S3Error::ClientConstructionFailed(err) => {
-                        Err(anyhow!("Client construction failed: {}", err))
-                    }
-                    S3Error::RetryFailed => Err(anyhow!("Retry failed")),
-                    S3Error::IoError(e) => Err(e.into()),
-                    S3Error::GetObjectError { .. } | S3Error::ListObjectsFailed(_) => {
-                        Ok(NextMessage::Pending)
-                    }
+            Some(Some(Err(e))) => match e {
```

Comment:
what does the double-Option mean? another good TODO to change that to an explicit enum

Reply:
I believe that the double-option is an an artifact of using `now_or_never()` on an async queue -- `recv()` yields an Option, and `now_or_never()` yields an Option of the thing it's called on.